lib/entity/src/iter.rs

```diff
@@ -10,16 +10,18 @@ where
     V: 'a,
 {
     pos: usize,
+    rev_pos: usize,
```

Comment:
You may want to model this on [`Enumerate`](https://github.com/rust-lang/rust/blob/5015fa346c1bf7e295fc16996ed0d3309c84f09a/src/libcore/iter/mod.rs#L1680) in the standard library which only uses one variable to track the position.

Reply:
I took a look, and it actually turned out to be straightforward to just use `Enumerate` in the implementation. Thanks for the suggestion!